portfolio/src/main/webapp/script.js

```diff
@@ -48,12 +48,20 @@ function newCommentBox(comment){
 }
 
 /**
- * Adds a greeting to the page using Promises
+ * Adds comments to the page using Promises
  */
 function loadComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  const commentContainer = document.getElementById('comment-container');
+  const parameters = location.search;
+  // Clear previous children
+  while(commentContainer.lastElementChild) {
+    commentContainer.removeChild(commentContainer.lastElementChild);
+  }
+
+  // Repopulate comment section
+  fetch('/data'+parameters).then(response => response.json()).then((comments) => {
```

Comment:
I wrote this before the discussion on using the Request class today. I know now that the Request class makes it more readable, but I think this method is pretty readable too? I guess the main difference is it does not state the name of the parameters and just passes them. Interested to know what you think.

Reply:
Correct. It is hard to tell what you parameters you are trying to pass. Assuming parameters was a nice simple string, then I guess its fine. In that case, Andrew would recommend template strings ```fetch(`/data?max-comments=${maxComments}`)```

With more complicated parameters, I would recommend Request.